### PR TITLE
common-updater-scripts: Support flake-compat repos

### PR DESCRIPTION
**Pull request by jtojnar**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/common-updater/scripts/update-source-version [hunk 1]
```diff
@@ -75,11 +79,39 @@ if [[ -z "$versionKey" ]]; then
     versionKey=version
 fi
 
+# Allow finding packages among flake outputs in repos using flake-compat.
+pname=$(nix-instantiate $systemArg --eval --strict -A "$attr.name" || echo)
+if [[ -z "$pname" ]]; then
```
Comment (SuperSandro2000) [inline pkgs/common-updater/scripts/update-source-version:84]: New test does not require quotes.

File: pkgs/common-updater/scripts/update-source-version [hunk 2]
```diff
@@ -75,11 +79,39 @@ if [[ -z "$versionKey" ]]; then
...
     if [[ ! -f "$nixFile" ]]; then
         die "Couldn't evaluate '$attr.meta.position' to locate the .nix file!"
     fi
+
+    # flake-compat will return paths in the Nix store, we need to correct for that.
+    possiblyOutPath=$(nix-instantiate $systemArg --eval -E "with $importTree; outPath" | tr -d '"')
+    if [[ -n "$possiblyOutPath" ]]; then
+        outPathEscaped=$(echo "$possiblyOutPath" | sed 's#[$^*\\.[|]#\\&#g')
+        pwdEscaped=$(echo "$PWD" | sed 's#[$^*\\.[|]#\\&#g')
+        nixFile=$(echo "$nixFile" | sed "s|^$outPathEscaped|$pwdEscaped|")
```
Comment (SuperSandro2000) [inline pkgs/common-updater/scripts/update-source-version:113]: This script should probably set -o pipefail.

**Review by romildo — APPROVED** [review]
Although I am not using flake, the diff looks good to me.
Verified that updating xfce, lxqt and mate still works with the commands:
``` console
$ nix-shell maintainers/scripts/update.nix --argstr path xfce
$ nix-shell maintainers/scripts/update.nix --argstr path lxqt
$ nix-shell maintainers/scripts/update.nix --argstr path mate
```

File: pkgs/common-updater/scripts/update-source-version (hunk 1 above)
Comment (jtojnar) [reply pkgs/common-updater/scripts/update-source-version:84]: I am aware but I prefer to use quotes everywhere to normalize their use.

File: pkgs/common-updater/scripts/update-source-version (hunk 2 above)
Comment (jtojnar) [reply pkgs/common-updater/scripts/update-source-version:113]: I do not think `echo` can fail in normal circumstances. Moreover, `pipefail` might be more trouble than it is worth: https://www.reddit.com/r/bash/comments/mivbcm/the_set_o_pipefail_is_not_a_best_practice/

File: pkgs/common-updater/scripts/update-source-version (hunk 1 above)
Comment (teto) [reply pkgs/common-updater/scripts/update-source-version:84]: plus when running shellcheck etc, it limits false positives.

File: pkgs/common-updater/scripts/update-source-version (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/common-updater/scripts/update-source-version:113]: Well thats the case for echo but anything other before a pipe can fail and the script wouldn't abort. ``set -eou pipefail`` should be set at the top of every sh file.

**Pull request merged**